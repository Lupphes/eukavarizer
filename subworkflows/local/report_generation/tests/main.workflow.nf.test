nextflow_workflow {

    name "Test Workflow REPORT_GENERATION"
    script "../main.nf"
    workflow "REPORT_GENERATION"
    config "subworkflows/local/report_generation/tests/nextflow.config"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "report_generation"

    test("Generate report with merged VCFs - stub") {
        options '-stub'

        when {
            workflow {
                """
                input[0] = 4932  // Yeast taxonomy ID
                input[1] = "$outputDir"
                input[2] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf', checkIfExists: true)
                input[3] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf', checkIfExists: true)
                input[4] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf.gz', checkIfExists: true)
                input[5] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf.gz.tbi', checkIfExists: true)
                input[6] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf', checkIfExists: true)
                input[7] = Channel.of(
                    [[id: 'caller1'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf.gz', checkIfExists: true)],
                    [[id: 'caller2'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test2.genome.vcf.gz', checkIfExists: true)]
                )
                input[8] = Channel.of(
                    [[id: 'caller1'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf.gz.tbi', checkIfExists: true)],
                    [[id: 'caller2'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test2.genome.vcf.gz.tbi', checkIfExists: true)]
                )
                input[9] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                input[10] = Channel.of(
                    [[id: 'sample1'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.stats', checkIfExists: true)]
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.report_file != null },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("Generate report with minimal inputs - stub") {
        options '-stub'

        when {
            workflow {
                """
                input[0] = 4932
                input[1] = "$outputDir"
                input[2] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf', checkIfExists: true)
                input[3] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf', checkIfExists: true)
                input[4] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf.gz', checkIfExists: true)
                input[5] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf.gz.tbi', checkIfExists: true)
                input[6] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf', checkIfExists: true)
                input[7] = Channel.of([[id: 'caller1'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf.gz', checkIfExists: true)])
                input[8] = Channel.of([[id: 'caller1'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test.genome.vcf.gz.tbi', checkIfExists: true)])
                input[9] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                input[10] = Channel.of([[id: 'sample1'], file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.stats', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.report_file != null }
            )
        }
    }
}
