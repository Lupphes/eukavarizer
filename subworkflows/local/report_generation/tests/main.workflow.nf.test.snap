{
    "Generate report with merged VCFs - stub": {
        "content": [
            {
                "0": [
                    "4932.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "4932.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "versions.yml:md5,2a162167379ff43e81df916b6d581d2e",
                    "versions.yml:md5,2a162167379ff43e81df916b6d581d2e"
                ],
                "report_file": [
                    "4932.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "4932.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,2a162167379ff43e81df916b6d581d2e",
                    "versions.yml:md5,2a162167379ff43e81df916b6d581d2e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-22T23:55:53.433314709"
    }
}